static-site/src/sections/group-members-page.tsx

```diff
@@ -0,0 +1,139 @@
+      let roles, members = [];
+      try {
+        const [ rolesResponse, membersResponse ] = await Promise.all([
+          fetch(uri`/groups/${groupName}/settings/roles`, headers),
+          fetch(uri`/groups/${groupName}/settings/members`, headers)
+        ]);
+        if (!rolesResponse.ok) {
+          throw new Error(`Fetching group roles failed: ${rolesResponse.status} ${rolesResponse.statusText}`)
+        }
+        if (!membersResponse.ok) {
+          throw new Error(`Fetching group members failed: ${membersResponse.status} ${membersResponse.statusText}`)
+        }
+        roles = await rolesResponse.json();
+        members = await membersResponse.json();
+      } catch (err) {
+        const errorMessage = (err as Error).message
+        if(!ignore) {
+          setErrorMessage({
+            title: "An error occurred when trying to fetch group membership data",
+            contents: errorMessage})
+        }
+      }
+      return {roles, members};
+    }
+
+    let ignore = false;
+    getGroupMembership(groupName).then(result => {
+      if (!ignore) {
+        setRoles(result.roles);
+        setMembers(result.members);
+      }
+    })
+    return () => {
+      ignore = true;
+    };
+  }, [groupName]);
+
+  return (
+    <GenericPage banner={errorMessage.title ? <ErrorBanner {...errorMessage} /> : undefined}>
+      <FlexGridRight>
+        <splashStyles.Button to={uri`/groups/${groupName}`}>
+          Return to {`"${groupName}"`} Page
+        </splashStyles.Button>
+      </FlexGridRight>
+      <MediumSpacer/>
+
+      <splashStyles.H2>
+        {`"${groupName}"`} Group Members
+      </splashStyles.H2>
+      <BigSpacer/>
+
+      {roles && members
+        ? <MembersTable roles={roles as string[]} members={members as GroupMember[]} />
+        : <splashStyles.H4>Fetching group members...</splashStyles.H4>}
+    </GenericPage>
+  )
+};
+
+const MembersTable = ({ roles, members }: { roles: string[], members: GroupMember[]}) => {
+  const sortedMembers = members.toSorted((a, b) => a.username.localeCompare(b.username));
+  function mostPrivilegedRole(memberRoles: string[]) {
+    // Assumes that the provided roles are listed in order of least to most privileged
+    return memberRoles.reduce((a, b) => roles.indexOf(a) > roles.indexOf(b) ? a : b);
+  }
+
+  return (
+    <CenteredContainer>
+      <div className="row">
+        <div className="col-6">
+          <splashStyles.CenteredFocusParagraph>
+            <strong>Username</strong>
+          </splashStyles.CenteredFocusParagraph>
+        </div>
+        <div className="col-6">
+          <splashStyles.CenteredFocusParagraph>
+            <strong>Roles</strong>
+          </splashStyles.CenteredFocusParagraph>
+        </div>
+      </div>
+
+    {sortedMembers.map((member) =>
+      <div className="row" key={member.username}>
+        <div className="col-6">
+          <splashStyles.CenteredFocusParagraph>
+            {member.username}
```

Comment:
Users can have odd usernames that make it hard to tell who it actually is. It would be cool if we could also list the "Name" attribute here since we have it on many users. This is what it looks like with e0e7b03f140fdda4be8bbc4a16a6e322172df6bf (I didn't think too much about the design, just wanted to see if adding the info was technically feasible):

![image](https://github.com/user-attachments/assets/0924e34a-00c3-4ff3-8b25-095a2f38c85d)

Reply:
> Users can have odd usernames that make it hard to tell who it actually is. It would be cool if we could also list the "Name" attribute here since we have it on many users.

Good point! I am just always hesitant to display any personal information because of data privacy issues.

Reply:
Thinking on this more, I'm going to punt on this for now. We can add more detailed user info later if we determine that it's necessary.